src/main/java/org/elasticsearch/search/aggregations/metrics/scripted/ScriptedMetricBuilder.java

```diff
@@ -187,20 +299,20 @@ protected void internalXContent(XContentBuilder builder, Params builderParams) t
             builder.map(reduceParams);
         }
         
-        if (initScript != null) {
-            builder.field(ScriptedMetricParser.INIT_SCRIPT, initScript);
+        if (initScriptString != null) {
+            builder.field(ScriptedMetricParser.INIT_SCRIPT, initScriptString);
         }
         
-        if (mapScript != null) {
-            builder.field(ScriptedMetricParser.MAP_SCRIPT, mapScript);
+        if (mapScriptString != null) {
+            builder.field(ScriptedMetricParser.MAP_SCRIPT, mapScriptString);
         }
         
-        if (combineScript != null) {
-            builder.field(ScriptedMetricParser.COMBINE_SCRIPT, combineScript);
+        if (combineScriptString != null) {
+            builder.field(ScriptedMetricParser.COMBINE_SCRIPT, combineScriptString);
         }
         
-        if (reduceScript != null) {
```

Comment:
In general we do validation in the Parser to keep all validation in one place and make the builders quite light-weight